lib/internal/Magento/Framework/Css/PreProcessor/Adapter/CssInliner.php

```diff
@@ -35,7 +39,8 @@ public function __construct(State $appState)
      */
     public function setHtml($html)
     {
-        $this->emogrifier->setHtml($html);
+        $this->emogrifier = EmogrifierCssInliner::fromHtml($html)
```

Comment:
I'd preserve the ability to call setters in any sequence. For this purpose it will be required to store passed values in properties and use them in the process method.
```suggestion
        $this->html = $html;
```